multiqc/modules/mirtrace/mirtrace.py

```diff
@@ -0,0 +1,381 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output files from miRTrace """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+
+import json
+import csv
+import random
+import string
+from multiqc import config
+from multiqc.plots import bargraph, linegraph
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='miRTrace',
+        anchor='mirtrace',
+        href='__TBD__',
+        info="is a quality control software for small RNA sequencing data. "\
+        "It is developed by the team of Marc Friedländer (KTH, Sweden).")
+
+        # Find and load miRTrace summary statistics table
+        self.summary_data = dict()
+        for f in self.find_log_files('mirtrace/summary'):
+            self.parse_summary(f)
+
+        # Find and load miRTrace read length table
+        self.length_data = dict()
+        for f in self.find_log_files('mirtrace/length'):
+            self.parse_length(f)
+
+        # Find and load miRTrace contamination statistics summary_table
+        self.contamination_data = dict()
+        for f in self.find_log_files('mirtrace/contaminationbasic'):
+            self.parse_contamination(f)
+
+        # Find and load miRTrace miRNA complexity table
+        self.complexity_data = dict()
+        for f in self.find_log_files('mirtrace/mirnacomplexity'):
+            self.parse_complexity(f)
+
+        # Filter to strip out ignored sample names
+        self.summary_data         =   self.ignore_samples(self.summary_data)
+        self.length_data          =   self.ignore_samples(self.length_data)
+        self.contamination_data   =   self.ignore_samples(self.contamination_data)
+        self.complexity_data      =   self.ignore_samples(self.complexity_data)
+
+        # Warning when no files are found
+        if len(self.summary_data) == 0:
```

Comment:
You should only have one of these checks and raise if there are none of any of the reports. We want to have the module run if we find anything, right? For example:

```python
if max(len(self.summary_data), len(self.length_data), len(self.contamination_data), len(self.complexity_data)) == 0:
    raise UserWarning
```

You can then use these checks later when adding the report sections, eg:

```python
if len(self.length_data) > 0:
    self.add_section (
        name = 'Read Length Distribution',
        anchor = 'mirtrace_length',
        plot = self.mirtrace_length_plot()
    )
```